library/core/build.rs

```diff
@@ -0,0 +1,16 @@
+// ferrocene addition
+fn main() {
+    let target = std::env::var("TARGET").expect("Cargo did not set the TARGET env var");
+
+    if !target.contains("-ferrocenecoretest") {
+        return;
+    }
+
+    match target.as_str() {
+        "aarch64-unknown-ferrocenecoretest" => {}
+        "thumbv7em-ferrocenecoretest-eabi" | "thumbv7em-ferrocenecoretest-eabihf" => {
+            println!("cargo::rustc-cfg=ferrocenecoretest")
+        }
```

Comment:
The fact that `cfg(ferrocenecoretest)` is not set for all coretest targets is really confusing. We should either set it for all of our targets (if appropriate) or find a better name for it.

Also, instead of duplicating this in all build scripts, we could add it in a single place [inside of bootstrap](https://github.com/ferrocene/ferrocene/blob/44d7fa90d6868978259eb4d0ff5f36525d51207a/src/bootstrap/src/core/build_steps/compile.rs#L451).